psh/date/date.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Phoenix-RTOS
+ *
+ * date - print or set the system date and time
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+char *wdaylookup[] = { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat" };
+
+char *monlookup[] = { "Jan", "Feb", "Mar", "Apr", "May", "June", "July", "Aug", "Sept", "Oct", "Nov", "Dec" };
+
+enum mode { SKIP,
+	HELP,
+	PRINT,
+	WRITE };
+
+static void psh_date_info(void)
+{
+	printf("print/set the system date and time");
+}
+
+
+static void psh_date_help(const char *prog)
+{
+	printf("Usage: %s [option] [epoch]\n", prog);
+	printf("  -h:  shows this help message\n");
+	printf("  -s:  set system time described by EPOCH (POSIX time format)\n");
+}
+
+
+static int psh_date_print(void)
+{
+	time_t t;
+	struct tm *tinfo;
+
+	time(&t);
+	tinfo = localtime(&t);
```

Comment:
```suggestion
	time_t t = time(NULL)
	struct tm *tinfo = localtime(&t);
```

Reply:
Left it as it is. Now there is some code executing before the time getting so it would be suitable to initialize it at the proper moment (for future we may want to print nanoseconds and then it could be handy to get time as close as possible to printing)